runtime/codegen/errors.go

```diff
@@ -31,3 +34,40 @@ func CatchPanics(r interface{}) error {
 	}
 	panic(r)
 }
+
+// JoinErrors is a backport of [errors.Join][1].
+//
+// TODO(mwhittaker): This is obsoleted by errors.Join. If we drop support for
+// go versions before go 1.20, we can remove this function.
+//
+// [1]: https://pkg.go.dev/errors#Join
+func JoinErrors(err1, err2 error) error {
+	return joinedErrors{err1, err2}
+}
+
+// joinedErrors is the combination of two errors.
+type joinedErrors struct {
+	err1 error
+	err2 error
+}
+
+// Error implements the error interface.
+func (j joinedErrors) Error() string {
+	return fmt.Sprintf("%v: %v", j.err1, j.err2)
+}
+
+// Is makes joinedErrors compatible with errors.Is.
+func (j joinedErrors) Is(err error) bool {
+	return errors.Is(j.err1, err)
```

Comment:
Is it intentional that errors.Is(j.err2, err) is not consulted? From what I can tell, the behavior here is currently incompatible with errors.Join().

Reply:
My understanding was `errors.Is` and `errors.As` call the `Is` and `As` method on every error returned by repeatedly calling `Unwrap`. I switched to go 1.20, as per your suggestion though, so I was able to delete this code :)